physics/src/modules/OceanBoundaryModule/include/TOPAZOcean.hpp

```diff
@@ -43,14 +45,19 @@ class TOPAZOcean : public IOceanBoundary, public Configured<TOPAZOcean> {
 private:
     // Updates the freezing point of an element
     void updateTf(size_t i, const TimestepTime& tst);
+
     // Since the configuration is global, it makes sense for the file path to
     // be static.
     static std::string filePath;
+    static bool doChecks;
 
     HField sstExt;
     HField sssExt;
 
     SlabOcean slabOcean;
+
+    IIceOceanHeatFlux* pIOHeatFlux;
+    IFreezingPoint* pFreezingPoint;
```

Comment:
These shouldn't be raw pointers. In fact, I don't think they should be here at all. Follow the method in the other ocean modules, and use the `Module` infrastructure.

Reply:
I copied what you did in ERA5Atmosphere with the IFluxCalculation reference #cargocultcoding I tried a bit with smart pointers, but I wasn't smart enough to make it work.